pkg/checker/functools.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package checker
+
+import (
+	"fmt"
+	"reflect"
```

Comment:
It is a good practice to cover all reflecty code with unit-tests.

Reply:
working on it